### DISCUSSION TITLE
How to handle manual user authentication on Flutter apps?

### DISCUSSION THREAD

**Discussion by FutureTree4839** [supabase/supabase#11393] [Questions]
**Quick summary of question:**

1. It seems that the best/quickest way to do manual user authentication (i.e. manually adding, editing and deleting users) on Supabase is to create a separate 'backend' application to handle these tasks. Is this true or is there any way to manually edit the auth.users table? 
2. Is it possible to run SQL queries to add new users or edit user passwords? (It seems it is possible to delete users via SQL - #250) 
3. If you add a user by email with the 'Invite' button and you don't have a user sign-up/administration process or website set up to direct the user to, a user is added to the auth.users table, but there seems to be no way to edit this user. Is there any way to edit or delete such a user?
4. On Firebase Authentication the functionality for manual creation, deletion and password resets of user accounts exists, so perhaps this might be a good feature request to have for Supabase. 
---

**Full question**

I'm trying to build a Flutter app with Supabase as the backend, and at various stages I need to _manually_ add some users to the app by specifying an email and a password (in order to give them 'admin' status on my app so that the can perform various 'admin' actions). Furthermore the only 'users' on my app will be these admin users, so there isn't any need for an automatic sign up process in the app. 

You can think of this setup/use case as being similar to a WordPress website where a few 'admin' users are manually created on the backend side with admin privileges' and there are further users who visit the website who have no privileges' and aren't signed up.

I'm trying to find out if there is quick way to handle the various bits of user administration on Supabase's backend for this use case (e.g. _manually_ creating a user, updating a user's password, deleting a user etc.). Ideally what I would like would be if there was some way to insert/edit/delete the rows on the auth.users table, but the auth schema is protected and doesn't allow manual entry of new rows to its subsequent tables or edits of existing ones (at least via the dashboard). 

If I want to add a new user to my auth table, Supabase does provide an 'Invite' button under the 'Authentication' tab on the backend dashboard, however that only sends a sign-up link to a specified email and I'd have to create a website to sign up a user once they click the link. The same goes for changing a password for a user. Since I'm building a Flutter app (and the target platforms are mobile-based) I'm trying to avoid having to set up a separate site to handle user authentication and administration. 

Another approach I can take to add a new user is by running a manual Dart or JavaScript command in order to sign a user up with a specified email or password (this is outlined in the docs [here](https://supabase.com/docs/reference/javascript/auth-signup)). But if I want to update a user's password I have to first manually sign the user in by running this command, then I have to run a subsequent command to update the user's password (as stated in the docs [here](https://supabase.com/docs/reference/javascript/auth-updateuser)). This can become a bit tedious because I also have to specify at some point the keys that point to my database for any of these commands to work. Perhaps the quickest way to run these commands might just be creating a separate app that connects to my Supabase database and only performs these user administration functions. 

Moreover, if for example, I attempt to add a user with a sign-up link by clicking the 'Invite' button on the Supabase backend (and I have no website implemented to handle the sign-up process), then a user (with no set password) is created on the auth.users table, but there is no way to sign this user in and update the user's password even by running any of the commands mentioned above. The problem is - you can't sign in the user with a command since there is no password set for the user and you can't update the user's password because the user is not signed in. This situation happened in one instance and now I have a user object on my auth.users table which I can't modify. In essence, this 'Invite' button only works if you have a corresponding website up and running to handle the user sign-up process. 

Basically for my use case it seems that in order to get user authentication to work (without having to run a bunch of commands) I have to create my own user 'backend' Flutter/Javascript application to handle the create/update/delete tasks I want for my users. 
Correspondingly on Firebase Authentication the functionality for manual creation and password resets of user accounts exists, so perhaps this might be a good feature request to have for Supabase.
